vhost-user-backend/src/handler.rs

```diff
@@ -279,6 +279,11 @@ where
             }
         }
 
+        let event_idx: bool = (self.acked_features & (1 << VIRTIO_RING_F_EVENT_IDX)) != 0;
+        for vring in self.vrings.iter_mut() {
+            vring.set_queue_event_idx(event_idx);
+        }
+        self.backend.set_event_idx(event_idx);
```

Comment:
Should we do this after setting the acked features in the backend?

I checked some backend and no one checks negotiated features (I don't know if we should fix this), but in theory it could avoid enabling event_idx if the feature is not negotiated.

Reply:
To be honest I'm not sure why we need an independent call (plus vring ones) instead of the backend checking for _F_EVENT_IDX set. What you say makes sense to me, but I can also see how a backend could simply check for acked features and then do not allow to modify it afterwards.

So I'm happy to other them the way is more convenient for sure. Should I reverse them?

Thanks!

Reply:
> To be honest I'm not sure why we need an independent call (plus vring ones) instead of the backend checking for _F_EVENT_IDX set. What you say makes sense to me, but I can also see how a backend could simply check for acked features and then do not allow to modify it afterwards.

I've been asking myself the same question, and I don't have an answer. :-) It would be nice to fix it, but not in this PR obviously.

> 
> So I'm happy to other them the way is more convenient for sure. Should I reverse them?

As you prefer, I also don't have a strong opinion. I think the best way is to avoid the independent call as you mention, but we can do later.

Thanks!

Reply:
Agreed, IMO the `backend.set_event_idx(event_idx)`API is redundant. And I prefer to be called before `        self.backend.acked_features(self.acked_features);`, but it doesn't really matter, the backend can always implement a nop `set_event_idx()`. I also this we should remove it in the future

Reply:
Okay, let's merge this and we can fix it in the future.